src/ORM/FieldType/DBInt.php

```diff
@@ -40,14 +40,14 @@ public function setValue(mixed $value, null|array|ModelData $record = null, bool
         parent::setValue($value, $record, $markChanged);
         // Cast string ints if they're valid
         if (is_string($this->value) && preg_match('/^-?\d+$/', $this->value)) {
-            // Ensure we can cast to int and back without loss of precision
-            // if not, keep the original value which will fail validation later
-            $stringIntValue = (string) (int) $value;
-            if ($stringIntValue !== $value) {
-                $this->value = $value;
+            // Use filter_var to validate if the string fits in a PHP integer
```

Comment:
```suggestion
            // Use filter_var to check if the string fits in a PHP integer
```
This is a nitpick and I won't hold up merging if it's not changed - but I'd like to avoid the word "validate" there since "validation" is a specific process that doesn't happen in this method.

Reply:
Updated